docs/rfcs/XeTile.md

```diff
@@ -25,60 +25,113 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
 ```mlir
-  %block = XeTile.init_tile %base_memref, [%tile_offset:2] :
+  %tile0 = XeTile.init_tile %base_memref, [%tile_offset:2] :
      memref<128x128xbf16> into tile<8x16xbf16>
 ```
 `init_tile` with memref of dynamic shape. The memref has a dynamic shape, so that its shape and strides have to be passed as runtime parameters to init_tile.
 ```mlir
-  %block = XeTile.init_tile %base_memref, [%tile_offset:2], [%base_shape:2], [%base_strides:2]:
+  %tile0 = XeTile.init_tile %base_memref, [%tile_offset:2], [%base_shape:2], [%base_strides:2]:
      memref<?x?xbf16> into tile<8x16xbf16>
 ```
  `init_tile` with an address for the base matrix. This form is to support the use case which doesn’t use a memref to describe the base matrix.
 ```mlir
-  %block = XeTile.init_tile %base_addr, [%tile_offset:2], [%base_shape:2], [%base_strides:2]:
+  %tile0 = XeTile.init_tile %base_addr, [%tile_offset:2], [%base_shape:2], [%base_strides:2]:
      i64 into tile<8x16xbf16>
 ```
+
+`init_tile` with an `order` to access the base matrix. The `order` attribute describes the order of the tile elements stored in the memory. "0" indicates the fastest-changing dimension. So if the base matrix is stored as row-major, the order is specified as [1, 0]. If the base matrix is stored as column major, the order is specified as [0, 1]. The default is row-major. The output tile carries the `order` attribute in its attribute set. 
+
+```mlir
+  #tile_attr = #xetile.tile_ttr<order = [0, 1]>
+  %tile0 = XeTile.init_tile %base_memref, [%tile_offset:2] {order = [0, 1]}:
+     memref<128x128xbf16> into tile<64x32xbf16, #tile_attr>
+```
+
+`init_tile` with an `inner_block` for 2D block access of the base matrix. The `inner_blocks` attribute describes the block size for each memory load and store operation when the tile is being loaded. The block size for load may be larger than the block size for MMA operation. The output tile carries the `inner_block` attribute in its attribute set. 
+
+```mlir
+  #tile_attr = #xetile.tile_ttr<inner_blocks=[16,16]>
+  %tile0 = XeTile.init_tile %base_memref, [%tile_offset:2] {inner_blocks=[16,16]}:
+     memref<128x128xbf16> into tile<64x32xbf16, #tile_attr>
+```
+
 With the tile date type, XeTile supports load_tile, prefetch_tile, and store_tile.
 
-`load_tile` loads a tile to a vector, which could be backed by a register region.
+`load_tile` loads a tile to a 2D vector, which could be backed by a register region.
 ```mlir
   %vector_a = XeTile.load_tile %tile_a :
      tile<64x64xbf16> into vector<64x64xb16>
 ```
 Attribute `transpose` specifies the dimensions being transposed along the load. It is commonly used for the GEMM on the backward path of DNN model, where one of input matrices needs to be transposed for matmul operation.
 ```mlir
-  %vector_a = XeTile.load_tile %tile_a {transpose = [1, 0]} :
+  %vector_a = XeTile.load_tile %tile_a {transpose = [0, 1]} :
```

Comment:
are we keeping the transpose?

Reply:
I am thinking removing it.  Need more discussions.

Reply:
I think the user may use both programming models: matmul (row_major, col_major), and matmul(row_major, row_maj^T). 
Below is one example for using row_major^T. However, when user uses this example, user will do the transpose explicitly and not combine with load. So it is XeTile's internal optimization which can take load and transpose and optimized to load with transpose at XeGPU level. 
https://github.com/openai/triton/blob/969c5bb1b828518c672376cf653f10ae13a0cfb4/test/TritonGPU/loop-pipeline.mlir#L552